src/js/jobs/components/Filters/StateFilter.tsx

```diff
@@ -30,11 +30,26 @@ const StyledStatusFilter = styled(SideBarSection)`
     z-index: 0;
 `;
 
-export const StateFilter = ({ counts, states, onUpdateJobStateFilter }) => {
-    const handleClick = value =>
-        onUpdateJobStateFilter(
+type StateFilterProps = {
+    counts: {
+        [state in JobState]?: {
+            [key: string]: number | null;
+        };
+    };
```

Comment:
`counts` typing should be defined in `types.ts` since it is part of the jobs response